File: src/app/material-advisor/src/app/models/knowledge-check/KnowledgeCheckTopicListItem.ts
import { GUID } from "@shared/types/GUID";
import { GeneratedFile } from "./GeneratedFile";

export interface KnowledgeCheckTopicListItem {
  id: GUID;
  name: string;
  startDate: Date;
  endDate?: Date;
  attemptsToVerifyCount: number;
  dataCount?: number;
}

File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/KnowledgeCheckTopicListItem.cs
namespace MaterialAdvisor.Application.Models.KnowledgeChecks;

public class KnowledgeCheckTopicListItem
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public int AttemptsToVerifyCount { get; set; }

    public int DataCount { get; set; }
}